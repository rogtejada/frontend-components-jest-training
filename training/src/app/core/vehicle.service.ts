import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { VehicleType, Vehicle } from '../shared/vehicle';
import { Paging } from '../shared/paging';

@Injectable({
  providedIn: 'root'
})
export class VehicleService {

  constructor(private httpClient: HttpClient) {
   }

   public searchVehicle(type: VehicleType, pageInfo: Paging): Vehicle[] {
    return null;
   }
}
