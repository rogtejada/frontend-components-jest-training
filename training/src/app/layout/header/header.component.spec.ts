import { Location } from '@angular/common';
import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { Router } from '@angular/router';
import { LayoutTestModuleModule } from 'src/app/test/layout-test-module/layout-test-module.module';
import { HeaderComponent } from './header.component';


describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;
  let router: Router;
  let location: Location;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ HeaderComponent ],
      imports: [
        LayoutTestModuleModule
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    router = TestBed.get(Router);
    location = TestBed.get(Location);
    fixture.detectChanges();

  });

  describe('Unit tests', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });
  });

  describe('DOM tests', () => {
    it('there are three navigation links', () => {
      const navLinks = fixture.debugElement.queryAll(By.css('.nav-link'));
      expect(navLinks.length).toBe(3);
    });
  });

  describe('Integration tests', () => {
    it('should navigate to the cars route', fakeAsync(() => {
      const carsLink = fixture.debugElement.queryAll(By.css('.nav-link'))[0];

      carsLink.nativeElement.click();
      fixture.whenStable().then(() => {
        expect(location.path()).toBe('/cars');
      });
    }));

    it('should navigate to the motorcycles route', () => {
      const link = fixture.debugElement.queryAll(By.css('.nav-link'))[1];

      link.nativeElement.click();
      fixture.whenStable().then(() => {
        expect(location.path()).toBe('/motorcycles');
      });
    });

    it('should navigate to the trucks route', () => {
      const link = fixture.debugElement.queryAll(By.css('.nav-link'))[2];

      link.nativeElement.click();
      fixture.whenStable().then(() => {
        expect(location.path()).toBe('/trucks');
      });
    });
  });

  describe('Snapshot tests', () => {
    it('Snapshot of html component=', () => {
      const header = fixture.debugElement.nativeElement.querySelector('nav');
      expect(header).toMatchSnapshot('HeaderComponent');
    });
  });
});
