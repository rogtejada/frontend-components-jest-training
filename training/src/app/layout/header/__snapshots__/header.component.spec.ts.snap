// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Snapshot tests Snapshot of html component=: HeaderComponent 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    class="navbar-brand"
    href="/"
    ng-reflect-router-link=""
  >
    Vehicle Store
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      class="navbar-nav"
    >
      <li
        class="nav-item"
        ng-reflect-router-link-active="active"
        routerlinkactive="active"
      >
        <a
          class="nav-link"
          href="/cars"
          ng-reflect-router-link="/cars"
        >
           Cars 
          <span
            class="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        ng-reflect-router-link-active="active"
        routerlinkactive="active"
      >
        <a
          class="nav-link"
          href="/motorcycles"
          ng-reflect-router-link="/motorcycles"
        >
           MotorCycles 
        </a>
      </li>
      <li
        class="nav-item"
        ng-reflect-router-link-active="active"
        routerlinkactive="active"
      >
        <a
          class="nav-link"
          href="/trucks"
          ng-reflect-router-link="/trucks"
        >
           Trucks 
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
